From 7e5c6204f4033e17da017f9d8eee4747f2f2536a Mon Sep 17 00:00:00 2001
From: "bfulgham@apple.com"
 <bfulgham@apple.com@268f45cc-cd09-0410-ab3c-d52691b4dbfc>
Date: Fri, 16 Dec 2016 18:32:27 +0000
Subject: [PATCH] Arguments called in wrong order
 https://bugs.webkit.org/show_bug.cgi?id=165923

Reviewed by Simon Fraser.

Source/WebCore:

Correct the ordering of several function calls.

* platform/graphics/GraphicsContext3D.cpp:
(WebCore::GraphicsContext3D::computeImageSizeInBytes): The 'bytesPerComponent' and 'componentsPerPixel'
variables were being passed in the wrong order.
* platform/network/CacheValidation.cpp: The 'last-modified' header string was being concatenated with
the following string, resulting in an incorrect 'last-modifiedproxy-authenticate' string value.
* svg/SVGToOTFFontConversion.cpp:
(WebCore::SVGToOTFFontConverter::firstGlyph): The ASSERT was assigning to the (by-value) input
parameter 'codePoint', rather than comparing it to the firstGlyph value's codePoint.

Tools:

The 'audioDeviceUIDs' array is being passed to the argument that expects video devices,
and the 'videoDeviceUIDs' array is being passed to the audo device argument. Fix this.

* TestWebKitAPI/Tests/WebKit2/UserMedia.cpp:
(TestWebKitAPI::decidePolicyForUserMediaPermissionRequestCallBack): The
order of arguments to WKUserMediaPermissionRequestAllow are flipped.


git-svn-id: http://svn.webkit.org/repository/webkit/trunk@209924 268f45cc-cd09-0410-ab3c-d52691b4dbfc
---
 .../platform/graphics/GraphicsContext3D.cpp        |  4 ++--
 .../WebCore/platform/network/CacheValidation.cpp   |  2 +-
 Source/WebCore/svg/SVGToOTFFontConversion.cpp      |  4 ++--
 Tools/TestWebKitAPI/Tests/WebKit2/UserMedia.cpp    |  3 ++-
 8 files changed, 50 insertions(+), 10 deletions(-)

diff --git a/Source/WebCore/platform/graphics/GraphicsContext3D.cpp b/Source/WebCore/platform/graphics/GraphicsContext3D.cpp
index 789f063..00a6a46 100644
--- a/Source/WebCore/platform/graphics/GraphicsContext3D.cpp
+++ b/Source/WebCore/platform/graphics/GraphicsContext3D.cpp
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2010 Apple Inc. All rights reserved.
+ * Copyright (C) 2010, 2016 Apple Inc. All rights reserved.
  * Copyright (C) 2010 Google Inc. All rights reserved.
  * Copyright (C) 2010 Mozilla Corporation. All rights reserved.
  *
@@ -329,7 +329,7 @@ GC3Denum GraphicsContext3D::computeImageSizeInBytes(GC3Denum format, GC3Denum ty
     if (width < 0 || height < 0)
         return GraphicsContext3D::INVALID_VALUE;
     unsigned int bytesPerComponent, componentsPerPixel;
-    if (!computeFormatAndTypeParameters(format, type, &bytesPerComponent, &componentsPerPixel))
+    if (!computeFormatAndTypeParameters(format, type, &componentsPerPixel, &bytesPerComponent))
         return GraphicsContext3D::INVALID_ENUM;
     if (!width || !height) {
         *imageSizeInBytes = 0;
diff --git a/Source/WebCore/platform/network/CacheValidation.cpp b/Source/WebCore/platform/network/CacheValidation.cpp
index b38e707..d49d475 100644
--- a/Source/WebCore/platform/network/CacheValidation.cpp
+++ b/Source/WebCore/platform/network/CacheValidation.cpp
@@ -46,7 +46,7 @@ const char* const headersToIgnoreAfterRevalidation[] = {
     "connection",
     "etag",
     "keep-alive",
-    "last-modified"
+    "last-modified",
     "proxy-authenticate",
     "proxy-connection",
     "trailer",
diff --git a/Source/WebCore/svg/SVGToOTFFontConversion.cpp b/Source/WebCore/svg/SVGToOTFFontConversion.cpp
index f0643d3..c708c5a 100644
--- a/Source/WebCore/svg/SVGToOTFFontConversion.cpp
+++ b/Source/WebCore/svg/SVGToOTFFontConversion.cpp
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2014-2015 Apple Inc. All rights reserved.
+ * Copyright (C) 2014-2016 Apple Inc. All rights reserved.
  *
  * Redistribution and use in source and binary forms, with or without
  * modification, are permitted provided that the following conditions
@@ -733,7 +733,7 @@ Glyph SVGToOTFFontConverter::firstGlyph(const Vector<Glyph, 1>& v, UChar32 codep
     auto codePoints = StringView(m_glyphs[v[0]].codepoints).codePoints();
     auto codePointsIterator = codePoints.begin();
     ASSERT(codePointsIterator != codePoints.end());
-    ASSERT(codepoint = *codePointsIterator);
+    ASSERT(codepoint == *codePointsIterator);
 #endif
     return v[0];
 }
diff --git a/Tools/TestWebKitAPI/Tests/WebKit2/UserMedia.cpp b/Tools/TestWebKitAPI/Tests/WebKit2/UserMedia.cpp
index 130aaef..fc7ade4 100644
--- a/Tools/TestWebKitAPI/Tests/WebKit2/UserMedia.cpp
+++ b/Tools/TestWebKitAPI/Tests/WebKit2/UserMedia.cpp
@@ -1,5 +1,6 @@
 /*
  * Copyright (C) 2014 Igalia S.L
+ * Copyright (C) 2016 Apple Inc. All rights reserved.
  *
  *  This library is free software; you can redistribute it and/or
  *  modify it under the terms of the GNU Lesser General Public
@@ -50,7 +51,7 @@ void decidePolicyForUserMediaPermissionRequestCallBack(WKPageRef, WKFrameRef, WK
         else
             audioDeviceUID = WKStringCreateWithUTF8CString("");
 
-        WKUserMediaPermissionRequestAllow(permissionRequest, videoDeviceUID.get(), audioDeviceUID.get());
+        WKUserMediaPermissionRequestAllow(permissionRequest, audioDeviceUID.get(), videoDeviceUID.get());
     }
 
     done = true;
-- 
2.9.3

